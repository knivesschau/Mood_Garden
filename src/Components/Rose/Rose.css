.Rose {
    text-align: center;
    border: 1px solid black;
    margin: 2em;
    padding: 1em;
}