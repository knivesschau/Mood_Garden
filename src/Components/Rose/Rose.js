import React, {Component} from 'react';
import moodGardenContext from '../../moodGardenContext';
import {Link} from 'react-router-dom';
import './Rose.css';

export default class Rose extends Component {
    
    static contextType = moodGardenContext;

    render() {
        const {id, rose, thorn, bud, color, entry_date} = this.props;

        return (
            <section className="Rose">

                <div className="View_Rose">
                    <Link to={`/roses/${id}`}>
                    <h2>{color} Rose Planted on {entry_date}</h2>
                    </Link>

                    <h3>Rose</h3>
                    <p>{rose}</p>

                    <h3>Thorn</h3>
                    <p>{thorn}</p>

                    <h3>Bud</h3>
                    <p>{bud}</p>

                    <button type="submit" id="edit">Edit</button>
                    <button type="submit" id="delete">Delete</button>

                </div>

            </section>
        );
    }
}
