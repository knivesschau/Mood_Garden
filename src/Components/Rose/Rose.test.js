import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Rose from './Rose';

describe ('Rose Component', () => {
    it ('renders the complete Rose Component', () => {
        const wrapper = shallow(<Rose/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})