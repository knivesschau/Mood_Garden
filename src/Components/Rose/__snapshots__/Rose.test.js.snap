// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rose Component renders the complete Rose Component 1`] = `
<section
  className="Rose"
>
  <div
    className="View_Rose"
  >
    <h2>
       Rose Planted on 
      Invalid Date
    </h2>
    <h3>
      Rose
    </h3>
    <p />
    <h3>
      Thorn
    </h3>
    <p />
    <h3>
      Bud
    </h3>
    <p />
    <Link
      to="/edit-rose/undefined"
    >
      <button
        id="edit"
        type="submit"
      >
        Edit Rose
      </button>
    </Link>
    <button
      id="delete"
      onClick={[Function]}
      type="button"
    >
      Delete Rose
    </button>
  </div>
</section>
`;
